### PR TITLE
fix truncate table

### PR DESCRIPTION
**Pull request by nautaa**
fix #113, #105

### CODE REVIEW

**Comment by jinmingjian** [comment]
@nautaa  welcome back, my boy! And you understand the idiom in store. It is great.

This impl is OK for serial execution but broken for concurrent working. 

1. It is better to first remove the meta then remove the data.
2. it is better to do meta removal in a transaction. I mark this as optional. Just have a try. there are some examples around the codes.

**Comment by nautaa** [comment]
@jinmingjian thanks for your advice, the transaction in `sled` is not support `iter()` method, so I use the `lock` in `PartStore`.

**Comment by jinmingjian** [comment]
@nautaa the usage of lock is interesting. Good catch if the transaction can not be supported (I will check further). Let's have this PR in! 

Thanks again!

File: crates/meta/src/store/parts.rs [hunk 1]
```diff
@@ -384,12 +384,34 @@ impl<'a> PartStore<'a> {
         Ok(())
     }
 
-    pub fn clear(&self)->MetaResult<()>{
-        //FIXME this clear stupid clear parts in the store
-        // self.tree_parts.clear().map_err(|_|MetaError::EntityDelError)?;
-        // self.tree_prids.clear().map_err(|_|MetaError::EntityDelError)?;
-        // self.tree_part_size.clear().map_err(|_|MetaError::EntityDelError)?;
+    pub fn clear(&self, _tid: Id, _cids: &[Id]) -> MetaResult<()> {
```
Comment (jinmingjian) [inline crates/meta/src/store/parts.rs:387]: _xxx id is for unread var naming idiom in Rust. So it is not good to appear in API's parameters. I personally pickup naming like this: ```cids``` in API, ```cids0```, ```cids1```... in the body for similar but different, or ```cids_ps`` (this is ```cids``` for parts store)

**Review by jinmingjian — APPROVED** [review]
I just leave the surprising naming for further fixs.

File: crates/meta/src/store/parts.rs (hunk 1 above)
Comment (nautaa) [reply crates/meta/src/store/parts.rs:387]: Thanks so much for your suggestion!

**Pull request merged**